cmd/operator-opamp-bridge/operator/client.go

```diff
@@ -165,36 +217,51 @@ func (c Client) Delete(name string, namespace string) error {
 	return c.k8sClient.Delete(ctx, &result)
 }
 
-func (c Client) ListInstances() ([]v1alpha1.OpenTelemetryCollector, error) {
+func (c Client) ListInstances() ([]v1beta1.OpenTelemetryCollector, error) {
 	ctx := context.Background()
-	result := v1alpha1.OpenTelemetryCollectorList{}
+
+	var instances []v1beta1.OpenTelemetryCollector
+
 	labelSelector := labels.NewSelector()
 	requirement, err := labels.NewRequirement(ManagedLabelKey, selection.In, []string{c.name, "true"})
 	if err != nil {
 		return nil, err
 	}
-	err = c.k8sClient.List(ctx, &result, client.MatchingLabelsSelector{Selector: labelSelector.Add(*requirement)})
-	if err != nil {
-		return nil, err
+	managedCollectorLabelSelector := client.MatchingLabelsSelector{Selector: labelSelector.Add(*requirement)}
+	reportingCollectorLabelMatcher := client.MatchingLabels{ReportingLabelKey: "true"}
+
+	managedCollectors, listErr := c.listCollectors(ctx, managedCollectorLabelSelector)
+	if listErr != nil {
+		return nil, listErr
 	}
-	reportingCollectors := v1alpha1.OpenTelemetryCollectorList{}
-	err = c.k8sClient.List(ctx, &reportingCollectors, client.MatchingLabels{
-		ReportingLabelKey: "true",
-	})
-	if err != nil {
-		return nil, err
+	instances = append(instances, managedCollectors.Items...)
+
+	reportingCollectors, listErr := c.listCollectors(ctx, reportingCollectorLabelMatcher)
+	if listErr != nil {
+		return nil, listErr
 	}
-	items := append(result.Items, reportingCollectors.Items...)
-	for i := range items {
-		items[i].SetManagedFields(nil)
+	instances = append(instances, reportingCollectors.Items...)
+
+	for i := range instances {
+		instances[i].SetManagedFields(nil)
 	}
 
-	return items, nil
+	return instances, nil
 }
 
-func (c Client) GetInstance(name string, namespace string) (*v1alpha1.OpenTelemetryCollector, error) {
+func (c Client) listCollectors(ctx context.Context, opts ...client.ListOption) (v1beta1.OpenTelemetryCollectorList, error) {
+	collectorsList := v1beta1.OpenTelemetryCollectorList{}
+	err := c.k8sClient.List(ctx, &collectorsList, opts...)
+	if err != nil {
+		return v1beta1.OpenTelemetryCollectorList{}, err
+	}
+	return collectorsList, nil
+}
```

Comment:
thought: i'm not sure we're getting much value from this as a helper and I think we should just stick with using the List directly given it is p much the same LOC.

Reply:
I liked what this did for the readability of the main `ListInstances` code to have separate, one-line assignments for the lists of managed collectors and reporting collectors. You're right though that it's a small change that's not saving many LOC, I'll re-work it.